lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -148,11 +148,23 @@ Distribution DistributionImplementation::operator + (const Distribution & other)
 
 Distribution DistributionImplementation::operator + (const DistributionImplementation & other) const
 {
-  if ((dimension_ != 1) || (other.dimension_ != 1)) throw NotYetImplementedException(HERE) << "In DistributionImplementation::operator + (const DistributionImplementation & other) const: can add 1D distributions only.";
-  Collection< Distribution > coll(2);
-  coll[0] = *this;
-  coll[1] = other.clone();
-  return new RandomMixture(coll);
+  if (dimension_ != other.dimension_) throw InvalidDimensionException(HERE) << "Can only sum distributions with the same dimension:" << dimension_ <<  " vs " << other.dimension_;
+
+  if ((dimension_ > 1) && (!hasIndependentCopula() || !other.hasIndependentCopula())) throw NotYetImplementedException(HERE) << "Can only sum distributions with independent copulas";
```

Comment:
If dimension_==1 then hasIndependentCopula() returns true (or it is a bug)

Reply:
Yes, then hasIndependentCopula is not even called and it falls through, what is your point ?

Reply:
My point is that in the univariate case, we can always sum distributions, and in the multivariate case, we have to check the copulas. I would prefer to have first a test to deal with the very common univariate case then to deal with the multivariate case, where the test would be reduced to the independent copulas on both side.